src/transformers/generation/utils.py

```diff
@@ -1147,6 +1147,21 @@ def _validate_model_kwargs(self, model_kwargs: Dict[str, Any]):
         # `prepare_inputs_for_generation` doesn't accept them, then a stricter check can be made ;)
         if "kwargs" in model_args or "model_kwargs" in model_args:
             model_args |= set(inspect.signature(self.forward).parameters)
+
+        # Generic composite models need information from its 2 components
+        if self.__class__.__name__ in {
+            "EncoderDecoderModel",
+            "VisionEncoderDecoderModel",
+            "SpeechEncoderDecoderModel",
+        }:
```

Comment:
```suggestion
        # Encoder-Decoder models may also need Encoder arguments from `model_kwargs`
        if self.config.is_encoder_decoder:
```

This issue is present in all encoder-decoder models :o 

Encoder kwargs may be needed in [this line](https://github.com/huggingface/transformers/blob/6112b1c6442aaf7affd2b0676a1cd4eee30c45cf/src/transformers/generation/utils.py#L644), when we prepare the encoder inputs -> we need to accept any Encoder argument

Reply:
But for concrete (not generic) encoder-decoder model, say `BartForConditionalGeneration` or `T5ForConditionalGeneration`, their forward already have all the arguments that their encoder components have.

I am not 100% sure if this holds in general. But I see the issue appears more in the generic encoder-decoder model, as the forward don't (actually can't) have all the arguments that (different) encoders will have.

Let me know if you still want me to commit this suggestion  though.

Reply:
> Let me know if you still want me to commit this suggestion though.

Yes please (may protect us from similar issues in the future) 🤗

Reply:
OK!